### ISSUE TITLE
pyproject fails to keep extra dependencies

### ISSUE THREAD

**Issue by greut** [dependabot/dependabot-core#2670]
**Package manager/ecosystem**
pip (pyproject and poetry)

**Manifest contents prior to update**

```toml
[tool.poetry.dependencies] 
boto3 = "^1.16.1"
boto3-stubs = {extras = ["autoscaling", "s3"], version = "^1.16.1"}
```

**Updated dependency**

To the next version of `boto3` or `boto3-stubs`, e.g. `1.16.2`

**What you expected to see, versus what you actually saw**

The `extras` is not taken into account and those are removed from the `poetry.lock`

**Images of the diff or a link to the PR, issue or logs**

```diff
 [package.dependencies]
-mypy-boto3-autoscaling = {version = "1.16.1.0", optional = true, markers = "extra == \"autoscaling\""}
-mypy-boto3-s3 = {version = "1.16.1.0", optional = true, markers = "extra == \"s3\""}
 typing-extensions = {version = "*", markers = "python_version < \"3.8\""}
```

Those should be kept.

**Comment by greut** [comment]
I was still using poetry 1.1.0, and it was probably fixed in poetry 1.1.1, https://github.com/python-poetry/poetry/pull/3036

**Comment by greut** [comment]
The latest pull requests were okay!
